### PR TITLE
Grtlv 130 hcsat where to export frontend

### PR DESCRIPTION
**Pull request by FoxJamie16**
HCSAT forms added to the 'Where to export' page, which appear once a user has interacted with the tool to open a comparison table, or returns to a previously populated comparison table. 

All user-facing changes are hidden behind the FEATURE_WTE_CSAT feature flag, which is to be removed once the 'Where to export' data has been updated (in progress). 

HCSAT ratings are saved in the core_csatuserfeedback table

### Workflow

- [x] Ticket exists in Jira: https://uktrade.atlassian.net/browse/GRTLV-130, https://uktrade.atlassian.net/browse/GRTLV-159
- [x] Jira ticket has the correct status.
- [x] A clear/description pull request messaged added.

### Reviewing help

- [x] Includes screenshot(s) - ideally before and after, but at least after

Before/FEATURE_WTE_CSAT=False:
where-to-export-csat-disabled
After+FEATURE_WTE_CSAT=True:
where-to-export-csat-enabled

### Housekeeping

- [x] I have checked that my PR is using the latest package versions of: great-components, directory-constants, directory-healthcheck, directory-validators, directory-components, directory-api-client, directory-ch-client, django-staff-sso-client, directory-forms-api-client, directory-sso-api-client, sigauth

### Merging

- [x] This PR can be merged by reviewers. (If unticked, please leave for the author to merge)

### CODE REVIEW

File: core/templates/core/compare_countries.html [hunk 1]
```diff
@@ -33,10 +33,52 @@ <h1>{{ page.title }}</h1>
...
+    <script src="{% static 'javascript/hcsat-feedback-form.js' %}"></script>
+    <script>
+    document.addEventListener('DOMContentLoaded', () => {
+        new CsatFormHandler('csat-form');
+    });
+    var list = document.getElementById("compare-markets");
+    var csat_form = document.getElementById("hcsat_section")
+
+    // Checks for table generated by React product/market modal, show hcsat if table has been generated
+    var MutationObserver = window.MutationObserver ||
```
Comment (FoxJamie16) [inline core/templates/core/compare_countries.html:62]: We only want to show the user's HCSAT once they've seen the comparison table which is entirely generated in React. A mutation observer is used on the containing div to check if a table with one or more data rows has been created, and if so then it counts the page as 'interacted with' and stops listening, showing HCSAT.

**Review by bobby-didcoding — COMMENTED** [review]
@FoxJamie16 You'll need to create some until tests for the models, forms and views. https://github.com/uktrade/great-cms/blob/8c7acbb2a927d7bd1a94d696b21a3816530034a4/tests/unit/export_academy/test_views.py#L271 is one of the tests for UKEA

**Comment by FoxJamie16** [comment]
> @FoxJamie16 You'll need to create some unit tests for the models, forms and views.
> 
> https://github.com/uktrade/great-cms/blob/8c7acbb2a927d7bd1a94d696b21a3816530034a4/tests/unit/export_academy/test_views.py#L271
> 
> is one of the tests for UKEA

Thanks Bobby, I'll take a look at the EAUK tests and make some!

**Review by bobby-didcoding — APPROVED** [review]
LGTM

**Pull request merged**